components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +133,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.having(FunctionType.TO_BOOLEAN, column, OperatorType.EQUALS_TO, "TRUE");
```

Comment:
I doubt this will work. The `HAVING` clause is meant to be a "where clause for aggregations" and that's not what you have here. I think it's a mistake that MetaModel even allows it, but that has to do with the common superclass "FunctionType" of both "AggregateFunction" and "ScalarFunction". It should only allow using a AggregateFunction here, but you're passing in a ScalarFunction (TO_BOOLEAN).